train_sd21.py

```diff
@@ -109,6 +109,15 @@
 check_min_version("0.27.0.dev0")
 
 
+def garbage_collection():
+    import gc
+
+    if torch.cuda.is_available():
+        torch.cuda.empty_cache()
+        torch.cuda.ipc_collect()
+    gc.collect()
```

Comment:
we can also call the torch.mps equivalents

this should probably be in `helpers/caching/memory.py` (just create it)

it can be called `reclaim_memory()` to match kohya's code.

Reply:
Nice!